scripts/ci/get_twister_opt.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2021 Intel Corporation
+
+# A script to generate twister options based on modified files.
+
+import re, os
+import sh
+import argparse
+import glob
+
+if "ZEPHYR_BASE" not in os.environ:
+    exit("$ZEPHYR_BASE environment variable undefined.")
+
+repository_path = os.environ['ZEPHYR_BASE']
+sh_special_args = {
+    '_tty_out': False,
+    '_cwd': repository_path
+}
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+                description="Generate twister argument files based on modified file")
+    parser.add_argument('-c', '--commits', default=None,
+            help="Commit range in the form: a..b")
+    return parser.parse_args()
+
+def find_archs(files):
+    archs = set()
+
+    for f in files:
+        if f.endswith(".rst") or f.endswith(".png") or f.endswith(".jpg"):
```

Comment:
The filename filter seems like it should be pulled out of each `find_<FOO>`.

Reply:
why? We do not want to trigger builds when only those files with rst,png and jpg have changed.